Sources/Paywalls/Components/PaywallButtonComponent.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PaywallButtonComponent.swift
+//
+//  Created by Jay Shortway on 02/10/2024.
+//
+// swiftlint:disable missing_docs
+
+import Foundation
+
+#if PAYWALL_COMPONENTS
+
+public extension PaywallComponent {
+
+    struct ButtonComponent: PaywallComponentBase {
+
+        let type: ComponentType
+        public let text: PaywallComponent.TextComponent
```

Comment:
This looks similar to the `LinkButton` where it has a`TextComponent` inside of it (which makes sense) but I totally forgot to delete the `LinkComponent` and mention when why it was build that way 😅 It was built for very early demo purpose and just reused `TextComponent` because easy.

However, I _think_ we are going to be leaning towards more of an protocol style instead of composition for components that have similar things here. 

I started doing this with the `StackableComponent` in https://github.com/RevenueCat/purchases-ios/pull/4324/files#diff-ec1da30bf93d1075c4c1985847a4762e27386eb3a6cb81a81dee288d50ba2fcc

So here, we could create a new protocol called like `TextableComponent` (names are hard???) that have the properties that are similar between `TextComponent` and `ButtonComponent`.

My reason for this is:
- Keeps JSON flatter
- Keeps components isolated
  - Ex: if we had a very specific property to `TextComponent`, we might not want `ButtonComponent` to have it (which actually happens in https://github.com/RevenueCat/purchases-ios/pull/4346 with selection state)

Reply:
That makes sense! However, are we going to allow people to give buttons arbitrary content, like SwiftUI and Jetpack Compose do? What about icons/images?

Reply:
This is... a great question 🤔 In the current spec, the button only has text. However...

Maybe a `ButtonComponent` should really be a container/stack and hold a list of components... and then the dashboard component selector can give options for like `TextButtonComponent` or `IconButtonComponent` that adds a button with those sub components in it already as a template?

Does that make sense? If so... we should bring this up to the Dan and Barbara 😇

Reply:
Yea that does make sense! And then things like "leading icon" vs "trailing icon" are just the same (horizontal) stack, but in a different order. I'll bring it up!

> Maybe a `ButtonComponent` should really be a container/stack

Yea, or it could have a single `StackComponent` child?

Reply:
Maybe making buttons behave like stacks (in whatever implementation) might be good to do regardless, to not limit our future selves. 🤔